Assets/MixedRealityToolkit.SDK/Experimental/Features/Dwell/Samples/BaseDwellSample.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dwell
+{
+    /// <summary>
+    /// Example script to demonstrate dwell visuals in sample scene
+    /// </summary>
+    public abstract class BaseDwellSample : MonoBehaviour
+    {
+        [SerializeField]
+        protected Image dwellVisualImage = null;
+
+        protected DwellHandler dwellHandler;
+
+        public bool isDwelling = false;
+
+        protected virtual void Awake()
+        {
+            dwellHandler = this.GetComponent<DwellHandler>();
+        }
+
+        public virtual void DwellStarted(IMixedRealityPointer pointer) { isDwelling = true; }
+
+        public virtual void DwellIntended(IMixedRealityPointer pointer) { }
+
+        public virtual void DwellCanceled(IMixedRealityPointer pointer) { isDwelling = false; }
+        
+        public virtual void DwellCompleted(IMixedRealityPointer pointer) { isDwelling = false; ButtonExecute(); }
+
+        public virtual void ButtonExecute() { }
```

Comment:
> I personally find this `ButtonExecute` method a little confusing. For me it would be easier to understand if we just kept it as `DwellCompleted` and the derived classes just override that method. It threw me off because I thought `ButtonExecute` was going to click a button or something, rather than just drive visuals corresponding to dwell completing.

yeah ButtonExecute is implementing the "what actions do i perform when this button is clicked", not doing anything related to dwell visuals :)
ButtonExecute is what will be bound to the unity button's onclick event handler.